### PR TITLE
New analyzer rule to replace count(*) with count(1) wherever it is found

### PR DESCRIPTION
**Pull request by zachmu**
This fixes https://github.com/dolthub/dolt/issues/5171

### CODE REVIEW

**Comment by zachmu** [comment]
Max, make sure you're ok with the changes to the query plans this introduces (adding a project in some cases)

**Comment by zachmu** [comment]
This may actually break some dolt enginetests, not going to merge until I figure out what's going on

**Comment by zachmu** [comment]
You can try out TestLogTableFunction, which is definitely broken by this change.

**Comment by zachmu** [comment]
I want to take another pass at this and make it show [] for an empty projection in the plan, but that was turning into too large a change.

It's definitely pruning columns as expected now, but a couple of the integration plans look a little off to me. Specifically, we have pruned like 6 columns but the indexes of other column expressions only shifted 2. That look right to you Max?

**Comment by max-hoffman** [comment]
The one with `FLQLP.LUEVY:17!null` -> `FLQLP.LUEVY:19!null`  had a "before" projection order 2, has an "after" order of 0. Math seems to add up.

File: enginetest/queries/query_plans.go [hunk 1]
```diff
@@ -10257,12 +10263,13 @@ WHERE
...
-			" └─ Table\n" +
-			"     ├─ name: NOXN3\n" +
-			"     └─ columns: [id brqp2 fftbj a7xo2 kbo7r ecdkm numk2 letoe ykssu fhcyt]\n" +
+		ExpectedPlan: "Project\n" +
+			" ├─ columns: [COUNT(1):0!null as COUNT(*)]\n" +
+			" └─ GroupBy\n" +
+			"     ├─ select: COUNT(1 (bigint))\n" +
+			"     ├─ group: \n" +
+			"     └─ Table\n" +
+			"         └─ name: NOXN3\n" +
```
Comment (max-hoffman) [inline enginetest/queries/query_plans.go:10272]: no table projection means reading all rows in this case?

**Review by max-hoffman — APPROVED** [review]
LGTM satisfying fix

File: enginetest/queries/query_plans.go (hunk 1 above)
Comment (zachmu) [reply enginetest/queries/query_plans.go:10272]: No, that's actually means it's projecting 0 columns, which is dumb. I want to fix that but it's just a lot bigger.

Comment (max-hoffman) [reply enginetest/queries/query_plans.go:10272]: how does it count the rows? does it just count a bunch of `sql.Row{nil}`s?

**Pull request merged**